### PR TITLE
PAYARA-3964 Allow configuration of MDB bean pool size with ActivationConfigProperty

### PR DESCRIPTION
**Pull request by smillidge**
Added support for singleton-bean-pool and bean-pool MDB deployment descriptor elements as ActivationConfigProperty values in MDB annotations

### CODE REVIEW

**Comment by arjantijms** [comment]
Jenkins test please

File: appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/deployment/annotation/handlers/MessageDrivenHandler.java [hunk 1]
```diff
@@ -114,7 +117,7 @@ protected EjbDescriptor createEjbDescriptor(String elementName,
      * Set Annotation information to Descriptor.
      * This method will also be invoked for an existing descriptor with
      * annotation as user may not specific a complete xml.
-     * @param ejbDesc
+     * @param ejbDescMaxWaitTimeInMillis
```
Comment (Pandrex247) [inline appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/deployment/annotation/handlers/MessageDrivenHandler.java]: ?
The param name _is_ `ejbDesc`

File: appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/deployment/annotation/handlers/MessageDrivenHandler.java [hunk 2]
```diff
@@ -142,6 +145,26 @@ protected HandlerProcessingResult setEjbDescriptorInfo(
             // xml override
             if (acProp.propertyName().equals("resourceAdapter")) {
                 ejbMsgBeanDesc.setResourceAdapterMid(envProp.getValue());
+            } else if (acProp.propertyName().equals("MaxPoolSize")) {
```
Comment (Pandrex247) [inline appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/deployment/annotation/handlers/MessageDrivenHandler.java]: Would a switch not be better?

**Comment by Pandrex247** [comment]
Nothing immediately _wrong_ from a quick glance, but a couple of quibbles

File: appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/deployment/annotation/handlers/MessageDrivenHandler.java (hunk 2 above)
Comment (jGauravGupta) [inline appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/deployment/annotation/handlers/MessageDrivenHandler.java]: The default value for these properties is defined in `appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/DescriptorConstants.java`. 
String constants should also be added for these properties name.

File: appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/deployment/annotation/handlers/MessageDrivenHandler.java (hunk 1 above)
Comment (arjantijms) [reply appserver/ejb/ejb-full-container/src/main/java/org/glassfish/ejb/mdb/deployment/annotation/handlers/MessageDrivenHandler.java]: Good catch, it's also not used for "MaxWaitTimeInMillis", so doesn't seem like a matter of forgetting to rename the method argument as well.

**Comment by Pandrex247** [comment]
Review comments PR: https://github.com/smillidge/Payara/pull/4

**Pull request merged**